src/Microsoft.Diagnostics.DebugServices.Implementation/CommandService.cs

```diff
@@ -568,14 +559,17 @@ Task<int> ICommandHandler.InvokeAsync(InvocationContext context)
             /// <summary>
             /// Returns true is the command is supported by the command filter. Calls the FilterInvokeAttribute marked method.
             /// </summary>
-            internal bool IsCommandSupported(Parser parser, IServiceProvider services) => _methodInfoFilter == null || (bool)Invoke(_methodInfoFilter, context: null, parser, services);
+            internal bool IsCommandSupported(RootCommand parser, IServiceProvider services) => _methodInfoFilter == null || (bool)Invoke(_methodInfoFilter, context: null, parser, services);
 
             /// <summary>
             /// Execute the command synchronously.
             /// </summary>
             /// <param name="context">invocation context</param>
             /// <param name="services">service provider</param>
-            internal void Invoke(InvocationContext context, IServiceProvider services) => Invoke(_methodInfo, context, context.Parser, services);
+            internal void Invoke(ParseResult context, IServiceProvider services) => Invoke(_methodInfo, context, (RootCommand)context.RootCommandResult.Command, services);
+
+            public override int Invoke(ParseResult parseResult)
+                => (int)Invoke(_methodInfo, parseResult, (RootCommand)parseResult.RootCommandResult.Command, null);
```

Comment:
The "services" argument can not be null here.  It is passed in as part of command execution and can't be part of the CommandHandler (like a field).  The old parser/ICommandHandler never called the Invoke override allowing the services instance to be passed through.   Now it looks like it needs to somehow be passed through to this override.